### PR TITLE
New Module nested styling #85

### PR DESCRIPTION
**Pull request by eric-martinez-sociomantic**
https://github.com/sociomantic-tsunami/nessie/issues/86
https://github.com/sociomantic-tsunami/nessie/issues/85

### CODE REVIEW

**Comment by eric-martinez-sociomantic** [comment]
screen shot 2017-11-15 at 13 03 58

File: src/Module/module.css [hunk 1]
```diff
@@ -1,84 +1,25 @@
...
-
-        &::before
-        {
-            @extend %module-highlight;
-            transition: height 200ms ease-in-out, opacity 200ms ease-in-out;
-            opacity:    1;
-        }
+        cursor:     pointer;
+        background: var(  --GS-11  );
+        transition: height 200ms ease-in-out, background-color 200ms ease-in-out;
```
Comment (conor-cafferkey-sociomantic) [inline src/Module/module.css]: The transition does not need to be defined here. It’s already defined for the `.header` on line 62.

(Also `height` is no longer needed... that must be left over from some earlier version of the Module.)

File: src/Module/module.css [hunk 2]
```diff
@@ -91,88 +32,35 @@ $moduleSpacing:  3rem;
...
         margin-bottom:      0;
 
-        &::before
-        {
-            transition: height 200ms ease-in-out, opacity 200ms ease-in-out;
-            opacity:    0;
-            height:     100%;
-        }
+        background: var(  --GS-11  );
+        transition: height 200ms ease-in-out, background-color 200ms ease-in-out;
```
Comment (conor-cafferkey-sociomantic) [inline src/Module/module.css]: Transition does not need to be defined here.

File: src/Module/module.css [hunk 3]
```diff
@@ -91,88 +32,35 @@ $moduleSpacing:  3rem;
...
+        height:     100%;
     }
 
-    &:hover > .header::before
+    &:hover > .header
     {
-        opacity:    1;
-        transition: height 200ms ease-in-out, opacity 200ms ease-in-out;
+        background: transparent;
+        transition: height 200ms ease-in-out, background-color 200ms ease-in-out;
```
Comment (conor-cafferkey-sociomantic) [inline src/Module/module.css]: The transition does not need to be defined here.

File: src/Module/module.css [hunk 4]
```diff
@@ -91,88 +32,35 @@ $moduleSpacing:  3rem;
...
-    {
-        transition:         height 200ms ease-in-out, opacity 200ms ease-in-out;
-        opacity:            1;
-
     }
 }
 
 .header
 {
+    background:         var(  --PC-WHITE  );
```
Comment (conor-cafferkey-sociomantic) [inline src/Module/module.css]: Redundant to set background to white here (you’re overriding it to be `transparent` above anyways).

File: src/Module/module.css [hunk 5]
```diff
@@ -91,88 +32,35 @@ $moduleSpacing:  3rem;
         padding-bottom:     3rem;
         margin-bottom:      0;
```
Comment (conor-cafferkey-sociomantic) [inline src/Module/module.css]: This `padding-bottom` and `margin-bottom`, and the `padding-bottom` above on line 28 are all redundant.

File: src/Module/module.css [hunk 6]
```diff
@@ -91,88 +32,35 @@ $moduleSpacing:  3rem;
...
 
-        &::before
-        {
-            transition: height 200ms ease-in-out, opacity 200ms ease-in-out;
-            opacity:    0;
-            height:     100%;
-        }
+        background: var(  --GS-11  );
+        transition: height 200ms ease-in-out, background-color 200ms ease-in-out;
+        height:     100%;
```
Comment (conor-cafferkey-sociomantic) [inline src/Module/module.css]: `height` is redundant here.

File: src/Module/index.jsx [hunk 7]
```diff
@@ -204,7 +198,6 @@ export default class Module extends Component
                     collapsed       : isCollapsible && isCollapsed,
                     error           : hasError,
                     moduleError     : hasModuleError,
-                    subtleHighlight : highlightType === 'subtle',
                     level           : headerLevel
```
Comment (conor-cafferkey-sociomantic) [inline src/Module/index.jsx]: `level` can be removed from here too, since it’s no longer used in the stylesheet.

**Comment by eric-martinez-sociomantic** [comment]
alles corrected @conor-cafferkey-sociomantic

File: src/Module/module.css [hunk 8]
```diff
@@ -17,48 +17,44 @@
...
     }
+
 }
 
 .moduleError
 {
-    & > .header
+    border-left: 0.5rem solid var( --AC-1 );
+
+    & > .header, .content
```
Comment (conor-cafferkey-sociomantic) [inline src/Module/module.css]: This _sort of_ works, but it’s not quite right...

The rule:
```
& > .header, .content
{
    padding-left: 2.5rem;
}
```

compiles to:

```
.module__moduleError > .module__header, 
.module__moduleError .module__content
{
    padding-left: 2.5rem;
}
```

But... what we really want is:
```
.module__moduleError > .module__header, 
.module__moduleError > .module__content
{
    padding-left: 2.5rem;
}
```

(Note the second `>`)

This is what we need to do:
```
& > .header, 
& > .content
{
    padding-left: 2.5rem;
}
```






Why does this matter? If you have something like this:

```
<Module hasModuleError>
    <Module>...</Module>
    <Module>...</Module>
</Module>
```

then we’d be applying the `padding-left: 2.5rem` not just to the `.content` div of the outer module, but also to the `.content` div of _every_ module nested inside it!

File: src/Module/index.jsx [hunk 9]
```diff
@@ -196,9 +196,7 @@ export default class Module extends Component
                 cssProps = { {
                     collapsible     : isCollapsible,
                     collapsed       : isCollapsible && isCollapsed,
-                    error           : hasError,
-                    moduleError     : hasModuleError,
```
Comment (conor-cafferkey-sociomantic) [inline src/Module/index.jsx]: Hmmm but... `hasError` and `hasModuleError` should be separate things with separate appearance.

The wanted behavior is the following:

- `hasError` adds an error icon in the header (our current implementation also requires `errorMessage` to be set too, in order for the icon to be shown)
- `hasModuleError` shows a red stripe on the left-hand side

The two props are independent from each other.

Now, admittedly, this is slightly confusing. But... it’s the current behavior and we need to keep it (for now at least)

**Comment by eric-martinez-sociomantic** [comment]
@conor-cafferkey-sociomantic corrected.

**Pull request merged**